src/components/ButtonBase/ButtonBase.style.ts

```diff
@@ -33,29 +33,22 @@ export const buttonBaseStyle = ({
 >) => (theme: Theme) => {
   const backGroundColor = defineBackgroundColor(theme, calculatedColor, type, childrenCount > 0);
   const isOutlined = !filled && !transparent;
-
-  const calculateButtonColor = () => {
-    if (type === 'link') {
-      const color = buttonConfig.types.link.color;
-
-      return theme.utils.getColor(color.name, color.shade);
-    }
-
-    if (isOutlined || transparent) {
-      return backGroundColor;
-    }
-
-    return theme.utils.getAAColorFromSwatches(calculatedColor.color, calculatedColor.shade);
-  };
+  const isBackgroundTransparent = isOutlined || transparent;
```

Comment:
I found these 2 lines are confusing. I mean, in the end I get what is happening, but I actually had to think about what is going on.

Reply:
I agree. This is something that will be partly fixed on v5 where the type of the button will be one variable instead of relying on 2 bool.

Reply:
v5 ticket [here](https://orfium.atlassian.net/browse/DS-326)